src/frontend/hooks/useKeepPlaystreaksInSync.ts

```diff
@@ -0,0 +1,36 @@
+import { questPlayStreakSyncState } from '@hyperplay/quests-ui'
+import { useEffect, useRef } from 'react'
+import { useQueryClient } from '@tanstack/react-query'
+
+export function useKeepPlaystreaksInSync(appName: string) {
+  const syncInitializedRef = useRef(false)
+  const queryClient = useQueryClient()
+
+  useEffect(() => {
+    return () => {
+      if (syncInitializedRef.current) {
+        questPlayStreakSyncState.clearAllTimers()
+      }
+    }
+  }, [])
+
+  useEffect(() => {
+    const initSync = async () => {
+      try {
+        questPlayStreakSyncState.init({
+          getQuests: window.api.getQuests,
+          getQuest: window.api.getQuest,
+          getUserPlayStreak: window.api.getUserPlayStreak,
+          syncPlaySession: window.api.syncPlaySession,
+          appQueryClient: queryClient
+        })
+
+        await questPlayStreakSyncState.keepProjectQuestsInSync(appName)
```

Comment:
`clearAllTimers` does not need to be called since the store and timers are destroyed when the overlay is closed, which is also when we should stop syncing play sessions